editor/utils/with-history/test/index.js

```diff
@@ -4,118 +4,138 @@
 import withHistory from '../';
 
 describe( 'withHistory', () => {
-	const counter = ( state = 0, { type } ) => (
-		type === 'INCREMENT' ? state + 1 : state
-	);
+	const counter = ( state = { count: 0 }, { type } ) => {
+		if ( type === 'INCREMENT' ) {
+			return { count: state.count + 1 };
+		}
+
+		if ( type === 'RESET' ) {
+			return { count: 0 };
+		}
+
+		return state;
+	};
+
+	const reducer = withHistory( counter, {
+		shouldOverwriteState: ( { type } ) => type === 'INCREMENT',
```

Comment:
These unit tests really only cover `shouldOverwriteState` usage, not standard usage without a custom overwrite handler.

Reply:
There's an action tested that does not overwrite but it could use some more tests indeed.